src/polymer/analyze-properties.ts

```diff
@@ -91,19 +92,15 @@ export function analyzeProperties(
 
         switch (propertyKey) {
           case 'type':
-            // If we've already got a type, then it was found in the jsdocs,
-            // and is canonical.
-            if (!prop.type) {
-              prop.type = esutil.objectKeyToString(propertyArg.value);
-              if (prop.type === undefined) {
-                prop.warnings.push(new Warning({
-                  code: 'invalid-property-type',
-                  message: 'Invalid type in property object.',
-                  severity: Severity.WARNING,
-                  sourceRange: document.sourceRangeForNode(propertyArg)!,
-                  parsedDocument: document
-                }));
-              }
+            prop.attributeType = esutil.objectKeyToString(propertyArg.value);
+            if (prop.attributeType === undefined) {
+              prop.warnings.push(new Warning({
```

Comment:
This seems like it'll introduce new warnings in cases where there was a type annotation, but an "invalid" Polymer type value. I'm not even sure the warning is correct, since things like `type: (v) => ...` are completely valid, yet will cause `undefined` to be returned by `objectKeyToString`.

Reply:
ok, I was wrong about arbitrary functions being "completely" valid. If Polymer sees something other than one of its supported types it just falls back to `Object`. Still not sure about the warning, since it's not uncommon to put another type there for documentation purposes, but that's a separate issue.

Reply:
Done (restored previous behavior, no warning if you have a `@type` override).